src/uct/ib/rdmacm/rdmacm_ep.c

```diff
@@ -8,10 +8,57 @@
 ucs_status_t uct_rdmacm_ep_resolve_addr(uct_rdmacm_ep_t *ep)
 {
     uct_rdmacm_iface_t *iface = ucs_derived_of(ep->super.super.iface, uct_rdmacm_iface_t);
+    ucs_status_t status;
+
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
+
+    status = uct_rdmacm_resolve_addr(ep->rdmacm_cm_id->cm_id,
+                                    (struct sockaddr *)&(ep->remote_addr),
+                                    UCS_MSEC_PER_SEC * iface->config.addr_resolve_timeout,
+                                    UCS_LOG_LEVEL_ERROR);
 
-    return uct_rdmacm_resolve_addr(iface->cm_id, (struct sockaddr *)&(ep->remote_addr),
-                                   UCS_MSEC_PER_SEC * iface->config.addr_resolve_timeout,
-                                   UCS_LOG_LEVEL_ERROR);
+    UCS_ASYNC_UNBLOCK(iface->super.worker->async);
+    return status;
+}
+
+ucs_status_t uct_rdmacm_ep_set_cm_id(uct_rdmacm_iface_t *iface, uct_rdmacm_ep_t *ep)
+{
+    ucs_status_t status;
+
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
+
+    /* create a cm_id for the client side */
+    if ((iface->num_cm_id_in_quota <= iface->config.cm_id_quota_size) &&
```

Comment:
why need to check that `iface->num_cm_id_in_quota <= iface->config.cm_id_quota_size` here?

Reply:
so that the amount of used cm_ids won't exceed the quota size.

Reply:
but num_cm_id_in_quota goes down when allocating new cm_id and not up, it's also compared to 0 for that reason